src/main/kotlin/org/rust/ide/inspections/RsDuplicatedTraitMethodBindingInspection.kt

```diff
@@ -0,0 +1,40 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.inspections
+
+import com.intellij.psi.PsiElementVisitor
+import org.rust.ide.refactoring.findBinding
+import org.rust.lang.core.psi.RsFunction
+import org.rust.lang.core.psi.RsVisitor
+import org.rust.lang.core.psi.ext.RsAbstractableOwner
+import org.rust.lang.core.psi.ext.owner
+
+class RsDuplicatedTraitMethodBindingInspection : RsLocalInspectionTool() {
+    override fun getDisplayName() = "Duplicated trait method parameter binding"
+
+    override fun buildVisitor(holder: RsProblemsHolder, isOnTheFly: Boolean): PsiElementVisitor =
+        object : RsVisitor() {
+            override fun visitFunction(function: RsFunction) {
+                if (function.owner !is RsAbstractableOwner.Trait) return
+                if (!function.isAbstract) return
+
+                val parameters = function.valueParameterList ?: return
+                val bindings = mutableSetOf<String>()
+                parameters.valueParameterList.forEach {
+                    val binding = it.pat?.findBinding() ?: return@forEach
+                    val name = binding.name ?: return@forEach
+                    if (name in bindings) {
+                        holder.registerProblem(
+                            binding,
+                            "Duplicated parameter name `$name`. Consider renaming it."
+                        )
+                    } else {
+                        bindings.add(name)
+                    }
+                }
```

Comment:
Using approach we don't highlight the first argument. And it's inconsistent with the corresponding error in `RsErrorAnnotator`. Also, it's possible that user should rename the first parameter instead of the second one. Let's better highlight all names with the same names

Reply:
I actually wanted this behaviour, but you're right that it's inconsistent. Btw maybe we should add a quick fix that would select and highlight the parameter for easier renaming? Is there anything existing like that? (I only found quick fixes that rename to a hard-coded string)

Reply:
> Btw maybe we should add a quick fix that would select and highlight the parameter for easier renaming?

Rename quick fix will be useful here, indeed. But let's add it in separate PR to merge this PR finally 😄 

> Is there anything existing like that? (I only found quick fixes that rename to a hard-coded string)

Unfortunately, it seems that we don't have such quick fix yet